examples/esp_idf/hello/main/app_main.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "esp_log.h"
+#include "nvs.h"
+#include "shell.h"
+#include "wifi.h"
+#if CONFIG_GOLIOTH_BLE_SERVICE_ENABLED
+#include "ble.h"
+#endif
+#include <golioth/golioth_sys.h>
+#include <golioth/client.h>
+#include <golioth/lightdb_state.h>
+#include <string.h>
+
+#define TAG "hello_espidf"
+
+// Given if/when the we have a connection to Golioth
+static golioth_sys_sem_t _connected_sem;
+
+static void on_client_event(struct golioth_client *client,
+                            enum golioth_client_event event,
+                            void *arg)
+{
+    bool is_connected = (event == GOLIOTH_CLIENT_EVENT_CONNECTED);
+    if (is_connected)
+    {
+        golioth_sys_sem_give(_connected_sem);
```

Comment:
Since this example is specific to ESP-IDF (unlike Golioth Basics, which runs on multiple platforms), let's use the ESP-IDF (in this case FreeRTOS) native APIs, instead of the `golioth_sys_*` abstractions. Same for the rest of this file, we shouldn't need to call any `golioth_sys_*` APIs.